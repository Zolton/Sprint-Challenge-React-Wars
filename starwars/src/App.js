import React, { useState, useEffect } from "react";
import StarWarsMap from "./components/StarWarsMap";
import "./App.css";
import axios from "axios";
import "semantic-ui-css/semantic.min.css";
import Loader from "./components/Loader";

const App = () => {
  // Try to think through what state you'll need for this app before starting. Then build out
  // the state properties here.

  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a
  // side effect in a component, you want to think about which state and/or props it should
  // sync up with, if any.

  const [starWarsChar, setStarWarsChar] = useState([]);
  //const [stretchGoal, setStretchGoal] = useState();

  useEffect(() => {
    axios
      .get("https://swapi.co/api/people/")
      .then(res => setStarWarsChar(res.data.results));
    //.then(console.log("starwarschar in axios", starWarsChar))
  }, []);

  // useEffect(() => {
  //   axios
  //     .get("https://dog.ceo/api/breeds/list/all")
  //     //.then(res=>(console.log("new stuff", res.data.message.hound)))
  //     .then(res=>setStretchGoal(res.data.message.hounds))
  //     //.then(res => setStarWarsChar(res.data.results));
  //   //.then(console.log("starwarschar in axios", starWarsChar))
  // }, []);

  return (
    <div className="App">
      <h1 className="Header">Star Wars is owed a paycheck and some royalties for this one</h1>
      {/* {console.log("from below return", starWarsChar)} */}
      {starWarsChar ? <StarWarsMap starWarsChar={starWarsChar} /> : <Loader />}
      {/* {stretchGoal ? stretchGoal.map(name=> name) : null} */}
    </div>
  );
};

export default App;
