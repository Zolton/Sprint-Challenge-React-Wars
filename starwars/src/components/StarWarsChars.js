import React from "react";

function StarWarsChars(props) {
  return (
    <>
      {/* {console.log("char props", props)} */}
      <h1>{props.indivSWChar.name}</h1>
      <h4>{props.indivSWChar.age}</h4>
    </>
  );
}

export default StarWarsChars;
