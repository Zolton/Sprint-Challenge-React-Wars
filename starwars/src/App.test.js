import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';
import StarWarsChars from './components/StarWarsChars';

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<StarWarsChars />', () => {
    it('should match snapshot', () => {
      const tree = renderer.create(<StarWarsChars />).toJSON();
  
      expect(tree).toMatchSnapshot();
    });
  });