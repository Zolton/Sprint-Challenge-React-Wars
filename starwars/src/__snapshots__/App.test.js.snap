// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="Header"
  >
    Star Wars is owed a paycheck
  </h1>
</div>
`;
